opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -789,18 +780,47 @@ def collect(
                 self._previous_max = current_max
                 self._previous_min = current_min
                 self._previous_sum = current_sum
+                self._previous_zero_count = current_zero_count
                 self._previous_positive = current_positive
                 self._previous_negative = current_negative
 
+                current_point = ExponentialHistogramDataPoint(
```

Comment:
it's not needed from correctness point of view, but it feels unnecessary to construct this data point & immediately throw it away, so I've put it into the if branch where it is used